lib/Doctrine/DBAL/Platforms/AbstractPlatform.php

```diff
@@ -2279,15 +2279,20 @@ public function getDefaultValueDeclarationSQL($field)
         if (isset($field['default'])) {
             $default = " DEFAULT '".$field['default']."'";
             if (isset($field['type'])) {
-                if (in_array((string) $field['type'], ["Integer", "BigInt", "SmallInt"])) {
+                $type = $field['type'];
+                if ($type instanceof Types\IntegerType ||
+                    $type instanceof Types\BigIntType ||
+                    $type instanceof Types\SmallIntType
+                ) {
                     $default = " DEFAULT ".$field['default'];
-                } elseif (in_array((string) $field['type'], ['DateTime', 'DateTimeTz']) && $field['default'] == $this->getCurrentTimestampSQL()) {
+                } elseif (($type instanceof Types\DateTimeType || $type instanceof Types\DateTimeTzType)
```

Comment:
Same here

Reply:
Indeed, and we probably should have added the immutable type here as well

Reply:
@lcobucci the immutable type extends the mutable type, so it's already ok :)